.ci/install.sh

```diff
@@ -37,12 +37,14 @@ python3 -m pip install -U pytest-timeout
 python3 -m pip install pyroma
 
 if [[ $(uname) != CYGWIN* ]]; then
-    python3 -m pip install numpy
+    # TODO Update condition when NumPy supports free-threading
+    if ! [[ "$GHA_PYTHON_VERSION" == "3.13-dev" ]]; then python3 -m pip install numpy ; fi
```

Comment:
We can change this to conditionally install `numpy` from `scientific-python-nightly-wheels`, which has nightly wheels that we can test against for free-threading.